systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/ManualRollingUpdateST.java

```diff
@@ -144,6 +145,50 @@ void testManualTriggeringRollingUpdate() {
         assertThat(received, is(sent));
     }
 
+    @Test
+    void testRollingUpdateOnNextReconciliationAfterClusterCAKeyDel() {
+        KafkaResource.kafkaPersistent(CLUSTER_NAME, 3, 3).done();
+        String topicName = KafkaTopicUtils.generateRandomNameOfTopic();
+        KafkaTopicResource.topic(CLUSTER_NAME, topicName, 2, 2).done();
+
+        KafkaUser user = KafkaUserResource.tlsUser(CLUSTER_NAME, USER_NAME).done();
+
+        KafkaClientsResource.deployKafkaClients(true, CLUSTER_NAME + "-" + Constants.KAFKA_CLIENTS, user).done();
+        final String defaultKafkaClientsPodName =
+            ResourceManager.kubeClient().listPodsByPrefixInName(CLUSTER_NAME + "-" + Constants.KAFKA_CLIENTS).get(0).getMetadata().getName();
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+            .withUsingPodName(defaultKafkaClientsPodName)
+            .withTopicName(topicName)
+            .withNamespaceName(NAMESPACE)
+            .withClusterName(CLUSTER_NAME)
+            .withMessageCount(MESSAGE_COUNT)
+            .withKafkaUsername(USER_NAME)
+            .build();
+
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+        Map<String, String> zkPods = StatefulSetUtils.ssSnapshot(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME));
+
+        int sent = internalKafkaClient.sendMessagesTls();
+        assertThat(sent, is(MESSAGE_COUNT));
+
+        String zkCrtBeforeAccident = kubeClient(NAMESPACE).getSecret(CLUSTER_NAME + "-zookeeper-nodes").getData().get(CLUSTER_NAME + "-zookeeper-0.crt");
```

Comment:
I would avoid using terms like accident, a delete in theory could be intentional.
best we cover cases of intentional or accidental removal and refer to it as zookeeperDeletedCert and kafkaDeletedCert ?

Reply:
I am aware as well that you didn't write these, but if you're moving the code, we should think about enhancing it slightly if we can.

Reply:
I didn't but I'm glad for comments to make the test better. So don't worry. :)